ql-qlf-plugin/ql-qlf-k6n10-plugin/synth_quicklogic.cc

```diff
@@ -0,0 +1,280 @@
+                inferBram = false;
+                continue;
+            }
+            if (args[argidx] == "-no_abc_opt") {
+                abcOpt = false;
+                continue;
+            }
+            
+            break;
+        }
+        extra_args(args, argidx, design);
+
+        if (!design->full_selection())
+            log_cmd_error("This command only operates on fully selected designs!\n");
+
+        log_header(design, "Executing SYNTH_QUICKLOGIC pass.\n");
+        log_push();
+
+        run_script(design, run_from, run_to);
+
+        log_pop();
+    }
+
+    void script() override
+    {
+        if (check_label("begin")) {
+            std::string readVelArgs;
+            readVelArgs = " +/quicklogic/" + family + "_cells_sim.v";
+            
+            run("read_verilog -lib -specify +/quicklogic/cells_sim.v" + readVelArgs);
+            run(stringf("hierarchy -check %s", help_mode ? "-top <top>" : top_opt.c_str()));
+        }
+
+        if (check_label("prepare"))  {
+            run("proc");
+            run("flatten");
+            run("opt_expr");
+            run("opt_clean");
+            run("deminout");
+            run("opt");
+        }
+
+        if (check_label("coarse")) {
+            run("opt_expr");
+            run("opt_clean");
+            run("check");
+            run("opt");
+            run("wreduce -keepdc");
+            run("peepopt");
+            run("pmuxtree");
+            run("opt_clean");
+
+            run("alumacc");
+            run("opt");
+            run("fsm");
+            run("opt -fast");
+            run("memory -nomap");
+            run("opt_clean");
+        }
+
+        if (check_label("map_bram", "(skip if -nobram)") && family == "qlf_k6n10" && inferBram) {
+            run("memory_bram -rules +/quicklogic/" + family + "_brams.txt");
+            run("techmap -map +/quicklogic/" + family + "_brams_map.v");
+        }
+
+        if (check_label("map_ffram")) {
+            run("opt -fast -mux_undef -undriven -fine");
+            run("memory_map -iattr -attr !ram_block -attr !rom_block -attr logic_block "
+                    "-attr syn_ramstyle=auto -attr syn_ramstyle=registers "
+                    "-attr syn_romstyle=auto -attr syn_romstyle=logic");
+            run("opt -undriven -fine");
+        }
+
+        if (check_label("map_gates")) {
+            if (inferAdder) {
+                run("techmap -map +/techmap.v -map +/quicklogic/" + family + "_arith_map.v");
+            } else {
+                run("techmap");
+            }
+            run("opt -fast");
+            run("opt_expr");
+            run("opt_merge");
+            run("opt_clean");
+            run("opt");
+        }
+
+        if (check_label("map_ffs")) {
+
+            std::string techMapArgs = " -map +/quicklogic/" + family + "_ffs_map.v";
+
+            if (family == "qlf_k6n10") {
+		run("dfflegalize -cell $_DFF_P_ 0 -cell $_DLATCH_P_ x");
+		}
+	    run("techmap " + techMapArgs);
```

Comment:
Does the K6N10 architecture support DLATCH? There is no techmap specified for DLATCH in the flow.

Reply:
Architecture doesn't support DLATCH, I'll be removing this.